tests/components/tado/test_service.py

```diff
@@ -0,0 +1,139 @@
+"""The serive tests for the tado platform."""
+import json
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.tado.const import (
+    ATTR_CONFIG_ENTRY,
+    ATTR_READING,
+    DOMAIN,
+    SERVICE_ADD_METER_READING,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+
+from .util import async_init_integration
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+async def test_has_services(
+    hass: HomeAssistant,
+) -> None:
+    """Test the existence of the Tado Service."""
+
+    await async_init_integration(hass)
+
+    assert hass.services.has_service(DOMAIN, SERVICE_ADD_METER_READING)
+
+
+async def test_add_meter_readings(
+    hass: HomeAssistant,
+) -> None:
+    """Test the add_meter_readings service."""
+
+    await async_init_integration(hass)
+
+    config_entry: MockConfigEntry = hass.config_entries.async_entries(DOMAIN)[0]
+    fixture: str = load_fixture("tado/add_readings_success.json")
+    with patch(
+        "homeassistant.components.tado.TadoConnector.set_meter_reading",
+        return_value=json.loads(fixture),
+    ):
+        response: None = await hass.services.async_call(
+            DOMAIN,
+            SERVICE_ADD_METER_READING,
+            service_data={
+                ATTR_CONFIG_ENTRY: config_entry.entry_id,
+                ATTR_READING: 1234,
+            },
+            blocking=True,
+        )
+        assert response is None
```

Comment:
In essence this is good for what will happen; you send the request and if no RequestExceptions are thrown it's all good. For the test it's a bit shallow, since we're not doing anything with the fixture here. If you use the `return_response` parameter ([see here](https://github.com/erwindouna/ha-core/blob/802f0da493cc3c59f0240ad6d80cc4dd3f561b98/homeassistant/core.py#L2162)), you can assert it against its response and effectively test this way the set_meter_reading went through the TadoConnector and PyTado.

Reply:
If there would be a `return_response` that would be correct, however there isn't any. Mocking the return_response just to satisfy the test seems ambiguous. A pytest will pass if it doesn't raise an exception, that's why originally I didn't add any assertion.

Reply:
Fair point. On second thought, this is indeed part of PyTado and should be fully handled there, which it does.